File: z_cs_py_yapilanlar/CKarZarar.cs
public class CKarZarar
{
    public dynamic V { get; set; }
    CTrader Trader = null;

    ~CKarZarar()
    {

    }

    public CKarZarar()
    {

    }

    public CKarZarar Initialize(dynamic Sistem, CTrader Trader)
    {
        this.Trader = Trader;
        this.V = Trader.V;

        return this;
    }

    public CKarZarar Reset(dynamic Sistem)
    {
        return this;
    }

    public int AnlikKarZararHesapla(dynamic Sistem, int BarIndex, string Type = "C")
    {
        int result = 0;

        int i = BarIndex;

        float AnlikFiyat = V[i].Close;

        bool AnlikKarZararHesaplaEnabled = Trader.Flags.AnlikKarZararHesaplaEnabled;

        string SonYon = Trader.Signals.SonYon;

        float SonFiyat = Trader.Signals.SonFiyat;

        int VarlikAdedSayisi = Trader.VarlikManager.VarlikAdedSayisi;

        if (AnlikKarZararHesaplaEnabled)
        {
            if (Type != "C")
            {
                if (Type == "O") AnlikFiyat = V[i].Open;
                else if (Type == "H") AnlikFiyat = V[i].High;
                else if (Type == "L") AnlikFiyat = V[i].Low;
            }

            if (SonYon == "A")
            {
                Trader.Status.KarZararPuan = (AnlikFiyat - SonFiyat);
                Trader.Status.KarZararFiyat = Trader.Status.KarZararPuan* VarlikAdedSayisi;

                Trader.Lists.KarZararPuanList[i] = Trader.Status.KarZararPuan;
                Trader.Lists.KarZararFiyatList[i] = Trader.Status.KarZararFiyat;

                Trader.Status.KarZararFiyatYuzde = 100.0f * Trader.Status.KarZararPuan / SonFiyat;
                Trader.Lists.KarZararFiyatYuzdeList[i] = Trader.Status.KarZararFiyatYuzde;
            }
            else if (SonYon == "S")
            {
                Trader.Status.KarZararPuan = (SonFiyat - AnlikFiyat);
                Trader.Status.KarZararFiyat = Trader.Status.KarZararPuan* VarlikAdedSayisi;

                Trader.Lists.KarZararPuanList[i] = Trader.Status.KarZararPuan;
                Trader.Lists.KarZararFiyatList[i] = Trader.Status.KarZararFiyat;

                Trader.Status.KarZararFiyatYuzde = 100.0f * Trader.Status.KarZararPuan / SonFiyat;
                Trader.Lists.KarZararFiyatYuzdeList[i] = Trader.Status.KarZararFiyatYuzde;
            }
            else if (SonYon == "F")
            {

            }

            if (Trader.Status.KarZararPuan > 0)
            {
                Trader.Status.KardaBarSayisi++;
                Trader.Status.ZarardaBarSayisi--;
            }
            else if (Trader.Status.KarZararPuan == 0)
            {
                Trader.Status.KardaBarSayisi = 0;
                Trader.Status.ZarardaBarSayisi = 0;
            }
            else
            {
                Trader.Status.KardaBarSayisi--;
                Trader.Status.ZarardaBarSayisi++;
            }
        }

        return result;
    }

}

File: z_cs_py_yapilanlar/CKarZarar.py
class CKarZarar:
    def __init__(self):
        self.V = None
        self.Trader = None

    def __del__(self):
        pass

    def initialize(self, Sistem, Trader):
        self.Trader = Trader
        self.V = Trader.V
        return self

    def reset(self, Sistem):
        return self

    def anlik_kar_zarar_hesapla(self, Sistem, BarIndex, Type="C"):
        result = 0
        i = BarIndex
        AnlikFiyat = self.V[i].Close

        AnlikKarZararHesaplaEnabled = self.Trader.Flags.AnlikKarZararHesaplaEnabled
        SonYon = self.Trader.Signals.SonYon
        SonFiyat = self.Trader.Signals.SonFiyat
        VarlikAdedSayisi = self.Trader.VarlikManager.VarlikAdedSayisi

        if AnlikKarZararHesaplaEnabled:
            if Type != "C":
                if Type == "O":
                    AnlikFiyat = self.V[i].Open
                elif Type == "H":
                    AnlikFiyat = self.V[i].High
                elif Type == "L":
                    AnlikFiyat = self.V[i].Low

            if SonYon == "A":
                self.Trader.Status.KarZararPuan = AnlikFiyat - SonFiyat
                self.Trader.Status.KarZararFiyat = self.Trader.Status.KarZararPuan * VarlikAdedSayisi
                self.Trader.Lists.KarZararPuanList[i] = self.Trader.Status.KarZararPuan
                self.Trader.Lists.KarZararFiyatList[i] = self.Trader.Status.KarZararFiyat
                if SonFiyat != 0:
                    self.Trader.Status.KarZararFiyatYuzde = 100.0 * self.Trader.Status.KarZararPuan / SonFiyat
                else:
                    self.Trader.Status.KarZararFiyatYuzde = 0.0
                self.Trader.Lists.KarZararFiyatYuzdeList[i] = self.Trader.Status.KarZararFiyatYuzde
            
            elif SonYon == "S":
                self.Trader.Status.KarZararPuan = SonFiyat - AnlikFiyat
                self.Trader.Status.KarZararFiyat = self.Trader.Status.KarZararPuan * VarlikAdedSayisi
                self.Trader.Lists.KarZararPuanList[i] = self.Trader.Status.KarZararPuan
                self.Trader.Lists.KarZararFiyatList[i] = self.Trader.Status.KarZararFiyat
                if SonFiyat != 0:
                    self.Trader.Status.KarZararFiyatYuzde = 100.0 * self.Trader.Status.KarZararPuan / SonFiyat
                else:
                    self.Trader.Status.KarZararFiyatYuzde = 0.0
                self.Trader.Lists.KarZararFiyatYuzdeList[i] = self.Trader.Status.KarZararFiyatYuzde

            if self.Trader.Status.KarZararPuan > 0:
                self.Trader.Status.KardaBarSayisi += 1
                self.Trader.Status.ZarardaBarSayisi -= 1
            elif self.Trader.Status.KarZararPuan == 0:
                self.Trader.Status.KardaBarSayisi = 0
                self.Trader.Status.ZarardaBarSayisi = 0
            else:
                self.Trader.Status.KardaBarSayisi -= 1
                self.Trader.Status.ZarardaBarSayisi += 1
                
        return result
